x-pack/plugins/security_solution/public/common/components/endpoint/responder/from_alerts/use_responder_action_data.test.ts

```diff
@@ -0,0 +1,283 @@
+        metadataApiMocks.responseProvider.metadataDetails.mockImplementation(() => {
+          throw createHttpFetchError('Not found', undefined, undefined, undefined, {
+            statusCode: 404,
+          });
+        });
+        const { result, waitForValueToChange } = renderHook();
+        await waitForValueToChange(() => result.current.tooltip);
+
+        expect(result.current).toEqual(
+          getExpectedResponderActionData({
+            isDisabled: true,
+            tooltip: NOT_FROM_ENDPOINT_HOST_TOOLTIP,
+          })
+        );
+      });
+
+      it('should show action as disabled with tooltip when host is found, but has a status of unenrolled', async () => {
+        const hostMetadata = {
+          ...metadataApiMocks.responseProvider.metadataDetails(),
+          host_status: HostStatus.UNENROLLED,
+        };
+        metadataApiMocks.responseProvider.metadataDetails.mockReturnValue(hostMetadata);
+
+        const { result, waitForValueToChange } = renderHook();
+        await waitForValueToChange(() => result.current.tooltip);
+
+        expect(result.current).toEqual(
+          getExpectedResponderActionData({
+            isDisabled: true,
+            tooltip: HOST_ENDPOINT_UNENROLLED_TOOLTIP,
+          })
+        );
+      });
+
+      it('should show action disabled if a metadata API error was encountered', async () => {
+        metadataApiMocks.responseProvider.metadataDetails.mockImplementation(() => {
+          throw createHttpFetchError('Server error', undefined, undefined, undefined, {
+            statusCode: 500,
+          });
+        });
+        const { result, waitForValueToChange } = renderHook();
+        await waitForValueToChange(() => result.current.tooltip);
+
+        expect(result.current).toEqual(
+          getExpectedResponderActionData({
+            isDisabled: true,
+            tooltip: METADATA_API_ERROR_TOOLTIP,
+          })
+        );
+      });
+    });
+  });
+
+  describe('useResponderActionData() hook', () => {
+    let hookProps: UseResponderActionDataProps;
+    let renderHook: () => RenderHookResult<UseResponderActionDataProps, ResponderActionData>;
+
+    beforeEach(() => {
+      endpointMetadataHttpMocks(appContextMock.coreStart.http);
+      hookProps = {
+        agentId: 'agent-123',
+        agentType: 'endpoint',
+        onClick: onClickMock,
+      };
+      renderHook = () => {
+        return appContextMock.renderHook<UseResponderActionDataProps, ResponderActionData>(() =>
+          useResponderActionData(hookProps)
+        );
+      };
+    });
+
+    it('should show action enabled when agentType is Endpoint and host is enabled', async () => {
+      const { result, waitForValueToChange } = renderHook();
+      await waitForValueToChange(() => result.current.isDisabled);
+
+      expect(result.current).toEqual(getExpectedResponderActionData());
+    });
+
+    it('should show action disabled if agent type is not Endpoint', () => {
```

Comment:
FYI: this is intentional. I had to use a valid `agentType` in the prop, so I chose `crowdstrikke` on line 257 - but it could have been sentinel one  and the result would have been the same. 

This test, however, will likely need to change once we introduce support for showing the `Respond` option on the SIEM Host page.